lib/protocol/collection/item_collection.ts

```diff
@@ -118,22 +118,34 @@ export class ItemCollection extends MutableCollection<SNItem> {
       const filteredCTMap = this.filteredMap[contentType]!;
       const sortedElements = this.sortedMap[contentType]!;
       /** If the element is deleted, or if it no longer exists in the primary map (because
-       * it was discarded without neccessarily being marked as deleted), it does not pass 
+       * it was discarded without neccessarily being marked as deleted), it does not pass
        * the filter. If no filter the element passes by default. */
       const passes = (element.deleted || !this.map[element.uuid]) ? false : (filter ? filter(element) : true);
```

Comment:
So it would seem that the filter can also access properties that are meant to be decrypted, and could cause exception. I believe the `errorDecrypting` check should occur before we access the `filter` function.

Here is the filter function web sets: https://github.com/standardnotes/web/blob/develop/app/assets/javascripts/views/notes/notes_view.ts#L317

And this is the underlying helper function it uses: https://github.com/standardnotes/web/blob/1c3530befa6300cce66ef271a4f6806d4b4abb76/app/assets/javascripts/views/notes/note_utils.ts#L14

This function accesses .pinned and .archived, which should actually just be regular properties, so undefined if errorDecrypting. So I'm not sure if this would throw exception. So I'm curious what it was exactly that caused mobile to throw exception?

Reply:
`const previousValue = (currentElement as any)[sortBy.key];` this line invokes 
```
get userModifiedDate() {
    const value = this.getAppDomainValue(AppDataField.UserModifiedDate);
    return new Date(value || this.updated_at);
}
```
And this will crash on the encrypted item. I believe it's a race condition that this happens on mobile and does not on the web.

Reply:
You know what, getters that access decrypted content should be moved into a property instead. I started heading towards this direction but appear to have missed this for `userModifiedDate`. So you should add this after this line similar to other properties: https://github.com/standardnotes/snjs/blob/004/lib/models/core/item.ts#L89

Reply:
Because this way we are in the safety check: https://github.com/standardnotes/snjs/blob/004/lib/models/core/item.ts#L83

Reply:
> So it would seem that the filter can also access properties that are meant to be decrypted, and could cause exception. I believe the errorDecrypting check should occur before we access the filter function.

I moved it to the top now.